README.md

```diff
@@ -1,101 +1,251 @@
-
+  * [Python] 3.6+
+  * The [Python] libraries: `numpy`, `matplotlib`, `toml` (must build from source), `shapely`, and `scipy`.
+
+### Detailed instructions for Ubuntu 18.04 LTS
+
+1. (Optional) Install Python 3.6+ (this comes natively in Ubuntu 18.04)
+2. Install `curl`:
+```bash
+sudo apt-get install curl
+```
+3. Install [rustup], the Rust toolchain (includes rustc, the compiler, and cargo, the package manager) from https://rustup.rs/ by running the following command and following on-screen instructions:
+```bash
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+```
+4. (Optional) Install `pip` for [Python]:
+```bash
+sudo apt-get install python3-pip
+```
+5. (Optional) Install [Python] libraries for making input files:
+```bash
+python3 -m pip install numpy matplotlib shapely scipy
+```
+6. (Optional) Install [Python] [TOML] library from source:
+```bash
+git clone https://github.com/uiri/toml.git
+cd toml
+python3 setup.py install
+```
+7. (Optional) Install software for [rcpr]:
+```bash
+sudo apt-get install gcc gfortran build-essential cmake liblapack-dev libblas-dev liblapacke-dev
+```
+8. Install `cargo`:
+```bash
+sudo apt-get install cargo
+```
+9. Build `rustBCA`:
+```bash
+git clone https://github.com/lcpp-org/rustBCA`
+cd rustBCA`
+cargo build --release
+```
+10. (Optional) Build `rustBCA` with optional dependencies, `hdf5` and/or `rcpr` (with your choice of backend: `openblas`, `netlib`, or `intel-mkl`):
+```bash
+cargo build --release --features cpr_rootfinder_netlib,hdf5_input
+cargo build --release --features cpr_rootfinder_openblas,hdf5_input
+cargo build --release --features cpr_rootfinder_intel_mkl,hdf5_input
+ ```
+11. `input.toml` is the input file - see [Usage](https://github.com/lcpp-org/RustBCA/wiki/Usage,-Input-File,-and-Output-Files) for more information
+12. Run the required tests using:
+```bash
+cargo test
+```
+13. (Optional) Run the required and optional tests for the desired backend(s):
+```bash
+cargo test --features cpr_rootfinder_netlib
+cargo test --features cpr_rootfinder_openblas
+cargo test --features cpr_rootfinder_intel_mkl
+```
+
+### Detailed instructions for Fedora 33
+
+Most of the ingredients for building `rustBCA` and running the [Python] helper
+scripts are available natively in the Fedora software repository, so the setup
+is relatively painless.
+
+The [Rust] toolchain can be aquired using:
+
+```bash
+sudo dnf install rust rust-src rust-std-static rust-analysis rust-gdb rust-lldb rustfmt
+```
+
+The (optional) [Python] packages can be obtained using:
+
+```bash
+sudo dnf install python3-numpy python3-scipy python3-matplotlib python3-toml python3-shapely
+```
+
+or, alternatively, using `pip3`.
+
+If the [rcpr] is desired, its probably also a good idea to install the following:
+
+```bash
+sudo dnf install gcc gcc-gfortran cmake lapack lapack-devel blas blas-devel
```

Comment:
rcpr is at a much, much earlier stage of development, so it's received a lot less attention from me w.r.t. building on multiple systems; does this set of dependencies allow it to compile on a clean installation of Fedora 33? That'd be a great piece of info to have.

Reply:
I can confirm that, with the above, I can build `rustBCA` with `--features cpr_rootfinder_netlib` and  `--features cpr_rootfinder_openblas`, but it's not clear to me that all of the above is necessary - they are just the Fedora equivalents of the packages listed for Ubuntu.

As far as I can tell, with the current build instructions, the full src of LAPACK and OpenBLAS are downloaded and built as part of their Rust bindings. The docs say that when building the (static) system libs can be provided, but this isn't on by default. The extremely long compilation times confirm this. 

On Fedora, static libs are also available and can be installed with:
```bash
sudo dnf install lapack-static openblas-static
```
but `Cargo.toml` would have to be updated to link to them.